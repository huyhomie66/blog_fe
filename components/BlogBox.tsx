import { Box, Text, Flex } from "@chakra-ui/react";
import Skeleton from "react-loading-skeleton";
import { FC, memo } from "react";
import { IBlog } from "@/types/blog";

interface Props extends IBlog {
  isLoading: boolean;
}

interface IField {
  title: string;
  content: string;
}

const BlogBox: FC<Props> = (props) => {
  const Field = ({ title, content }: IField) => {
    if (props.isLoading) {
      return <Skeleton data-testid="skeleton-loader" />;
    }
    return (
      <Flex>
        <Text fontWeight={"bold"}>.{title}: </Text>
        <Text>{content}</Text>
      </Flex>
    );
  };
  return (
    <Box border={"solid 0.5px "} borderRadius={4} w="70vw">
      <Field title={"Title"} content={props.title} />
      <Field
        title={"Created At"}
        content={new Date(props.createdAt).toLocaleDateString("en-GB")}
      />
      <Field title={"Created By"} content={props.createdBy} />
      <Field title={"Content"} content={props.content} />
    </Box>
  );
};

export default memo(BlogBox);
