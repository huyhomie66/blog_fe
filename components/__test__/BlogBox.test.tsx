// components/__test__/BlogBox.test.tsx
import { render } from "@testing-library/react";
import BlogBox from "../BlogBox";

describe("BlogBox", () => {
  it("renders BlogBox unchanged", () => {
    const blogData = {
      id: "1",
      title: "Test Blog",
      createdAt: new Date(),
      createdBy: "Test User",
      content: "Test content",
      isLoading: false
    };

    const { container } = render(<BlogBox {...blogData} />);
    expect(container).toMatchSnapshot();
  });
});
