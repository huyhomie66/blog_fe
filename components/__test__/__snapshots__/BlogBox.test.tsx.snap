// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogBox displays a skeleton loader when isLoading is true 1`] = `
<div>
  <div
    class="css-v5drb2"
  >
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
      >
        ‌
      </span>
      <br />
    </span>
  </div>
</div>
`;

exports[`BlogBox displays the blog data when isLoading is false 1`] = `
<div>
  <div
    class="css-v5drb2"
  >
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Title
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test Blog
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Created At
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        19/02/2023
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Created By
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test User
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Content
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test content
      </p>
    </div>
  </div>
</div>
`;

exports[`BlogBox renders BlogBox unchanged 1`] = `
<div>
  <div
    class="css-v5drb2"
  >
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Title
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test Blog
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Created At
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        20/02/2023
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Created By
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test User
      </p>
    </div>
    <div
      class="css-k008qs"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        .
        Content
        : 
      </p>
      <p
        class="chakra-text css-0"
      >
        Test content
      </p>
    </div>
  </div>
</div>
`;
